spec/lib/search_spec.rb

```diff
@@ -2221,6 +2223,36 @@ def indexed_post(*args)
       expect(results.more_full_page_results).to eq(nil)
     end
 
+    it 'returns more results flag' do
+      results = Search.execute('hello', search_type: :header)
+      expect(results.posts.length).to eq(Search.per_facet)
+      expect(results.more_posts).to eq(nil) # not 6 posts yet
+
+      post6 = Fabricate(:post, raw: 'hello post #6')
+
+      results = Search.execute('hello', search_type: :header)
+      expect(results.posts.length).to eq(Search.per_facet)
+      expect(results.more_posts).to eq(true)
+    end
+  end
+
+  context 'header in-topic search' do
+    let!(:topic) { Fabricate(:topic, title: 'This is a topic with a bunch of posts') }
+    let!(:post1) { Fabricate(:post, topic: topic, raw: 'hola amiga') }
+    let!(:post2) { Fabricate(:post, topic: topic, raw: 'hola amigo') }
+    let!(:post3) { Fabricate(:post, topic: topic, raw: 'hola chica') }
+    let!(:post4) { Fabricate(:post, topic: topic, raw: 'hola chico') }
+    let!(:post5) { Fabricate(:post, topic: topic, raw: 'hola hermana') }
+    let!(:post6) { Fabricate(:post, topic: topic, raw: 'hola hermano') }
+    let!(:post7) { Fabricate(:post, topic: topic, raw: 'hola chiquito') }
+
+    it 'does not trigger pagination early' do
```

Comment:
A note here but I don't quite understand what `early` means in this context.

Reply:
So, we have two types of pagination in search: `per_filter` (50) and `per_facet` (5). Full page search always uses `per_filter`, and header uses `per_facet` in all cases but one (searching posts in a topic).  This test case ensures that the `per_facet` limit is not used when searching posts in a topic.